### DISCUSSION TITLE
Setup my User and password for mqtt client

### DISCUSSION THREAD

**Discussion by khoih-prog** [khoih-prog/AsyncMQTT_Generic#15] [Q&A]
From @Gendiaaa 

https://github.com/khoih-prog/AsyncMQTT_Generic/discussions/3#discussioncomment-3892004

===

hello Sir ,
i want to know how to  in your Example for Esp32 please

**Comment by khoih-prog** [comment]
Currently you don't need `UserName` and `Password` for this test MQTT broker

https://github.com/khoih-prog/AsyncMQTT_Generic/blob/aab1f0927e6b036656b4dbc6f423af7ada2cc255/examples/ESP32/FullyFeaturedSSL_ESP32/FullyFeaturedSSL_ESP32.ino#L33

You can change to another MQTT server / broker, and you have to know how to do that according to every broker. You can even create and run local MQTT broker in your computer.

Do some Google research.

**Reply by Gendiaaa** [reply]
thank you very much for your replay 
Sir i already have broker setuped in aws server was looking for library to use it with esp32 to control dosing peristaltic  pumps 
 mqttClient.setCredentials(mqttUserName, mqttPwd);
this was the function i got it 
but for the code example it uses freertos and TimerHandle with symaphore as i can understand .. i tried to setup it but once my pump function runs it block the code and the esp restarted also there was no proper time to let the esp32 connect again to wifi it tries to connect but WiFi lost connection it was hard for me to understand it i think i need just more time to study the method

**Comment by khoih-prog** [comment]
If you're not familiar with complex code, I suggest you don't use Async. 

Just check

- [**WiFiMQTTSecureAWS** example](https://github.com/khoih-prog/MQTTPubSubClient_Generic/tree/main/examples/WiFi/WiFiMQTTSecureAWS) of [**MQTTPubSubClient_Generic** library](https://github.com/khoih-prog/MQTTPubSubClient_Generic)

Moreover, you're using ESP32, it's better you ask the question in original [**MQTTPubSubClient**](https://github.com/hideakitai/MQTTPubSubClient) or [**arduino-mqtt**](https://github.com/256dpi/arduino-mqtt) library.

Here the support is targeted for boards other than ESP32/ESP8266.
